### PR TITLE
DYN-6857-homepage-preferences

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

This PR addresses a jira request for improvement of the HomePage: `As a Dynamo user, I want Grid/List view option picked on Dynamo Home persist across sessions` https://jira.autodesk.com/browse/DYN-6857.

> As a Dynamo user, I want Grid/List view option picked on Dynamo Home persist across sessions.
> 
> - Needs to be serialized
> - Needs to persist across session
> - Do not require user to set initial value
> - Preferred to be migrated with newer version of Dynamo launch
>
> We have examples in Preferences Settings just like this, e.g. Dynamo Window Default size, Dynamo Library Default size, etc. Although this setting is not expected to be exposed in settings panel
> 

Works with the following HomePage changes: https://github.com/DynamoDS/DynamoHome/pull/13

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

- added a single serialized property that unpacks into a dictionary of items which are sent to the front-end HomePage each time Dynamo is started
- the single property would allow any number of settings to be stored in the future without the need to alter the back-end side
- currently 2 properties are being sent from the front-end
- - recentPageViewMode 
- - samplesViewMode 

### Reviewers

@QilongTang 
@Amoursol 

### FYIs

@reddyashish

### CODE REVIEW

**Comment by QilongTang** [comment]
Oops, I think I accidentally merged the wrong PR..

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 1]
```diff
@@ -395,6 +395,21 @@ public bool IsBackgroundPreviewActive
         /// </summary>
         public bool UseHardwareAcceleration { get; set; }
 
+        /// <summary>
+        /// Persistence for Dynamo HomePage
+        /// </summary>
+        [XmlIgnore]
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:401]: 👍🏻

**Comment by QilongTang** [comment]
@dnenov Would you re-propose this PR?

**Comment by reddyashish** [comment]
I will create a new PR for this one as Deyan is going on leave.

**Comment by QilongTang** [comment]
> I will create a new PR for this one as Deyan is going on leave.

Thank you, appreciate it!

**Pull request merged**